Anima/diffusion/mcm_estimator/animaMCMEstimatorImageFilter.hxx

```diff
@@ -348,21 +348,74 @@ MCMEstimatorImageFilter<InputPixelType, OutputPixelType>
             m_MCMCreators[i]->SetConcentrationBounds(kappaLowerBound,kappaUpperBound);
     }
     
-    // Setting up coarse grids for kappa and nu initial search (useful for NODDI)
-    m_FractionCoarseGrid.resize(m_CoarseGridSize);
-    m_KappaCoarseGrid.resize(m_CoarseGridSize);
+    // Switch over compartment types to setup coarse grid initialization
+    switch (m_CompartmentType)
+    {
+        case NODDI:
+            this->ComputeNODDIExtraAxonalAndKappaCoarseGrids();
+            break;
+
+        case DDI:
+            itkExceptionMacro("DDI coarse grid not implemented yet");
```

Comment:
Yes we can use the NODDI initialisation also here. I am hoping it is going to be ok.